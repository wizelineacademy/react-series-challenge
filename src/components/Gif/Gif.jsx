import React from 'react';
import PropTypes from 'prop-types';

import {
  GifContainer,
  GifTitle,
  GifImage,
  GifLikeContainer,
} from './Gif.style';

import likeContainer from './favorite-heart-like-container.png';
import likeSelected from './favorite-heart-like-selected.png';

const Gif = ({ object, addFavorite, removeFavorite }) => (
  <GifContainer>
    <GifTitle>{object.title}</GifTitle>
    <GifImage src={object.embed_url} title={object.title} />
    {
      object.isFavorite ?
        (
          <GifLikeContainer
            src={likeSelected}
            alt="Remove from favorite"
            onClick={() => removeFavorite({ payload: { gif: object } })}
          />
        ) :
        (
          <GifLikeContainer
            src={likeContainer}
            alt="Add to favorite"
            onClick={() => addFavorite({ payload: { gif: object } })}
          />
        )
    }
  </GifContainer>
);

Gif.propTypes = {
  object: PropTypes.shape({
    embed_url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
    isFavorite: PropTypes.bool.isRequired,
  }).isRequired,
  addFavorite: PropTypes.func.isRequired,
  removeFavorite: PropTypes.func.isRequired,
};

export default Gif;

