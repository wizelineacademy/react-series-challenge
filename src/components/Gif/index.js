import { bindActionCreators } from 'redux';
import { connect } from 'react-redux'
import gif from '../../actions/gif'
import React from 'react'

export const Gif = ({ src, alt, onClick, className, gif, favorite }) => {
    return <img alt={alt} src={src} onClick={onClick.bind(null, gif)} className={ favorite ? `${className} favorite` : className } />
}


const mapDispatchToProps = dispatch => {
    /**
     *
     */
    const { creators: { clickGif: onClick } } = gif
    /**
     *
     */
    return bindActionCreators({ onClick }, dispatch)
}

export default connect(null, mapDispatchToProps)(Gif)