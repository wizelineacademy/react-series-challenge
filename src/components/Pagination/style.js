import styled from 'styled-components'

export const PaginationButton = styled.span`
  color: black;
  background-color: white;
  padding: 8px 16px;
  text-decoration: none;
  transition: background-color .3s;
  border: 1px solid #ddd;
  cursor: pointer;
  margin-top: 30px;
  ${ props => props.active && 
    `background-color: #4CAF50;
    color: white;
    border: 1px solid #4CAF50;
    cursor: default;
    `
  }
`