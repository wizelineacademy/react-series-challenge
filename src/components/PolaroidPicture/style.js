import styled from 'styled-components'
import { Link } from 'react-router-dom';


export const PolaroidListItem = styled.li`
  position: relative;
  width: ${props => props.isSingleView ? '500px;' : '200px;'}
  height: ${props => props.isSingleView ? '650px;' : '250px;'}
  display: inline-block;
  padding: 10px;
  margin: 10px;
  background-color: white;
  box-shadow: 7px 7px 4px black;
  border: 1px solid #000
  img {
    max-width: 100%;
    height: 80%;
  }
`
export const PolaroidLink = styled(Link)`
  color: white;
`

export const PolaroidStar = styled.div`
  background-image: url('/favorited.ico');
  background-size: contain;
  width: 50px;
  height: 50px;
  position: absolute;
  top: -25px;
  right: calc(50% - 25px);
`

export const PolaroidButton = styled.button`
  margin: 5px;
  padding: 10px 19px;
  margin: 0px 10px -10px 0px;
  border-radius: 10px;
  font-family: 'Pacifico', cursive;
  font-size: 17px;
  color: #FFF;
  text-decoration: none; 
  border: none;
  outline:none;
  cursor: pointer;
  display: none;
  ${ props => props.color === 'blue' ?
      `
        background-color: #3498DB;
        text-shadow: 0px -2px #2980B9;
      `
      :
      `
        background-color: #82BF56;
        text-shadow: 0px -2px #669644;
        float: right;
      `
  }
  :active {
    transform: translate(0px,5px);
    -webkit-transform: translate(0px,5px);
    border-bottom: 1px solid;
  }
`

export const PolaroidLoading = styled.span`
  position: absolute;
  width: 200px;
  height: 200px;
  left: 10px;
  background-image: url("/loader.svg");
  background-size: contain;
  background-color: black;
`