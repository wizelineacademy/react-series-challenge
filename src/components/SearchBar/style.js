import styled from 'styled-components'

export const Container = styled.div `
  display: flex;
  justify-content: center;
  margin-bottom: 70px;
`

export const SearchBox = styled.div `
  position: absolute;
  input {
    border: none;
    background: none;
    z-index: 1;
    width: 50px;
    height: 50px;
    transition: all .25s ease-in .25s;
    color: transparent;
    font-size: 1.25rem;
    line-height:50px;
    &:hover {
      cursor: pointer;
      &:focus {
        cursor: text;
      }
      + span {
        background: rgba(255,255,255,0.2);
      }
    }
    &:focus {
      width: 200px;
      padding: 0 10px;
      outline: none;
      color: black;
      background:none;
      color: white;
      + span {
        width: 200px;
        &::before {
          width:2px;
          opacity: 0;
          transition: all .25s ease-in;
        }
      }
    }
    + span {
      z-index: -1;
      position: absolute;
      border: 2px solid white;
      top: 0;
      width: 50px;
      height: 50px;
      transition: all .25s ease-in .25s;
      border-radius: 50px;
      left: 0;
      &::before {
        transition: all .25s ease-in .5s;
        transform-origin: left top;
        content: '';
        position: absolute;
        width: 20px;
        height: 7px;
        border-radius: 5px;
        background: white;
        transform: rotate(45deg) translate(43px, 10px);
      }
    }
  }
`