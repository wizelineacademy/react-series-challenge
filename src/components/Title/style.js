import styled from 'styled-components'

export const BurningEffect = styled.h1`
  color: #fff;
  text-shadow: 0px -1px 4px white, 0px -2px 10px yellow, 0px -10px 20px #ff8000, 0px -18px 40px red;
  font: 2em 'BlackJackRegular';
`

export const NeonEffect = styled.h1`
  color: #fff;
  text-shadow: 0 0 5px #fff, 0 0 10px #fff, 0 0 15px #fff, 0 0 20px #ff2d95, 0 0 30px #ff2d95, 0 0 40px #ff2d95, 0 0 50px #ff2d95, 0 0 75px #ff2d95;
  letter-spacing: 5px;
  font: 3em 'Miso';
`