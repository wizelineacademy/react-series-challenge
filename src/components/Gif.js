import React, {
    Component
} from "react";
import {
    addFav,
    withdrawFav
} from "../Containers/FavPage/action"
import {
    connect
} from 'react-redux'

import {Wraper, ImgStyled} from "./Gif.style";

class Gifs extends Component {
    constructor(props) {
        super(props);
        this.state = {
            buttonText: "Fav",
            fav: false
        }
        this.onClick = this.onClick.bind(this);
    }

    componentDidMount() {
        if (this.props.gif.length > 0) {
            this.props.gif.map((image) => {
                if (image.id === this.props.id) {
                    this.setState({
                        fav: true,
                        buttonText: "NotFav"
                    });
                }
            });
        }
    }

    onClick() {
        if (this.props.gif.length > 0) {
            this.props.gif.map((image) => {
                if (image.id === this.props.id) {
                    this.setState({
                        fav: true
                    });
                }
            });
            if(this.state.fav){
                this.props.withdrawFav(this.props.id);
                this.setState({
                    fav: false,
                    buttonText:"Fav"
                });
            } else{ this.props.addFav({
                id: this.props.id,
                url: this.props.url
            });
            this.setState({
                fav: true,
                buttonText:"NotFav"
            });
            }
        } else {
            this.props.addFav({
                id: this.props.id,
                url: this.props.url
            });
            this.setState({
                fav: true,
                buttonText:"NotFav"
            });
        }

    }

    render() {
        return ( 
            < React.Fragment >
            <Wraper onClick = {this.onClick}>
                <ImgStyled src={require("./"+this.state.buttonText+".png")}   />
                <img 
                alt = ""
                src = {this.props.url}
                key = {this.props.id}
                />
                </Wraper>
            </ React.Fragment>
        );
    }
}

const mapStateToProps = state => {
    return {
        gif: state.fav.favGifs
    }
}

const mapDispatchToProps = {
    addFav,
    withdrawFav
}

export default connect(mapStateToProps, mapDispatchToProps)(Gifs);