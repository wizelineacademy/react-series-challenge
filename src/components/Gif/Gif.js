import React from 'react';
import {MasonryItem, Overlay, ImageWrapper, LinkStyled, GifImg, GifTitle} from "./Gif.styled";

const getPreviewGifUrl = (gif) => {
  if (gif.images && gif.images.original) {
    return gif.images.original.url;
  } else {
    return '';
  }
};

const handleFavButton = (gif, addGif, removeGif, isFavorited) => {
  if (!isFavorited) {
    addGif(gif);
  } else {
    removeGif(gif.id);
  }
};

export const Gif = ({ gif, fullWidth, addGif, removeGif, isFavorited, isInDetail }) => (
  <MasonryItem fullWidth={fullWidth}>
      <ImageWrapper>
        <GifImg isFavorited={isFavorited} src={getPreviewGifUrl(gif)} alt={gif.title} />
        <Overlay>
          <GifTitle>Title: {gif.title}</GifTitle>
          <button id="fav-btn" onClick={() => handleFavButton(gif, addGif, removeGif, isFavorited)}>
            {!isFavorited ? 'Add to Favorites': 'Remove from favorites'}
          </button>
          { !isInDetail ? (
            <LinkStyled to={{pathname: `/detail/${gif.id}`}}>
              View detail
            </LinkStyled>) : ''
          }
        </Overlay>
      </ImageWrapper>
  </MasonryItem>
)