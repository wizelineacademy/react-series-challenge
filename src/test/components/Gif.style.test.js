import React from 'react'
import renderer from 'react-test-renderer'
import { Wraper } from "../../components/Gif.style"

test('Styled gif works', () => {
  const tree = renderer.create(<Wraper />).toJSON()
  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('width', '200px')
})