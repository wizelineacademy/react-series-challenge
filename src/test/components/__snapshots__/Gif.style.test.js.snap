// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled gif works 1`] = `
<div
  className="sc-bdVaJa jbzoYE"
/>
`;
